test/v2/client/test_client_setup.go

```diff
@@ -0,0 +1,240 @@
+
+	return config, nil
+}
+
+// GetTestClient is the same as GetClient, but also performs a check to ensure that the test is not
+// running in a CI environment. If using a TestClient in a unit test, it is critical to use this method
+// to ensure that the test is not running in a CI environment.
+func GetTestClient(t *testing.T, configPath string) *TestClient {
+	skipInCI(t)
+	c, err := GetClient(configPath)
+	require.NoError(t, err)
+	return c
+}
+
+// GetClient returns a TestClient instance, creating one if it does not exist.
+// This uses a global static client... this is icky, but it takes ~1 minute
+// to read the SRS points, so it's the lesser of two evils to keep it around.
+func GetClient(configPath string) (*TestClient, error) {
+	clientLock.Lock()
+	defer clientLock.Unlock()
+
+	if client, ok := clientMap[configPath]; ok {
+		return client, nil
+	}
+
+	testConfig, err := GetConfig(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get config: %w", err)
+	}
+
+	if len(clientMap) == 0 {
+		// do one time setup
+
+		var loggerConfig common.LoggerConfig
+		if os.Getenv("CI") != "" {
+			loggerConfig = common.DefaultLoggerConfig()
+		} else {
+			loggerConfig = common.DefaultConsoleLoggerConfig()
+		}
+
+		testLogger, err := common.NewLogger(loggerConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create logger: %w", err)
+		}
+		logger = testLogger
+
+		// only do this stuff once
+		err = setupFilesystem(logger, testConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to setup filesystem: %w", err)
+		}
+
+		testMetrics := newTestClientMetrics(logger, testConfig.MetricsPort)
+		metrics = testMetrics
+		testMetrics.start()
+	}
+
+	client, err := NewTestClient(logger, metrics, testConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create test client: %w", err)
+	}
+
+	clientMap[configPath] = client
+
+	return client, nil
+}
+
+func skipInCI(t *testing.T) {
+	if os.Getenv("CI") != "" {
+		t.Skip("Skipping test in CI environment")
+	}
+}
+
+func setupFilesystem(logger logging.Logger, config *TestClientConfig) error {
+	// Create the test data directory if it does not exist
+	err := os.MkdirAll(config.TestDataPath, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create test data directory: %w", err)
+	}
+
+	// Create the SRS directories if they do not exist
+	srsPath, err := config.Path(SRSPath)
+	if err != nil {
+		return fmt.Errorf("failed to get SRS path: %w", err)
+	}
+	err = os.MkdirAll(srsPath, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create SRS directory: %w", err)
+	}
+	srsTablesPath, err := config.Path(SRSPathSRSTables)
+	if err != nil {
+		return fmt.Errorf("failed to get SRS tables path: %w", err)
+	}
+	err = os.MkdirAll(srsTablesPath, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create SRS tables directory: %w", err)
+	}
+
+	// If any of the srs files do not exist, download them.
+	filePath, err := config.Path(SRSPathG1)
```

Comment:
nit: looks like the logic  "check if a file exists, and download a file from an endpoint if not" could be extracted to a helper method

Reply:
done